scripts/log2json

```diff
@@ -0,0 +1,194 @@
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+
+# This script takes the combined logfile from the tests and summarizes
+# the results as JSON that can be imported into a log analyzer.
+#
+# The results are similar to this:
+#
+# [
+#         {
+#             "group": "logs-daily-iso",
+#             "logfile": "INFO: install_log = /var/tmp/kstest-anaconda-conf.2022_03_04-03_50_44.h13p_9sy/virt-install.log",
+#             "start_time": 1646383844.0,
+#             "summary": [
+#                 "Logs are in /var/tmp/kstest-anaconda-conf.2022_03_04-03_50_44.h13p_9sy",
+#                 "Disk image(s) at /var/tmp/kstest-anaconda-conf.2022_03_04-03_50_44.h13p_9sy/disk-a.img,cache=unsafe",
+#                 "Results are in /var/tmp/kstest-anaconda-conf.2022_03_04-03_50_44.h13p_9sy"
+#             ],
+#             "result": "INFO: RESULT:anaconda-conf:ea89963f4984:SUCCESS:test done",
+#             "success": true,
+#             "name": "anaconda-conf",
+#             "end_time": 1646385568.0,
+#             "elapsed_time": 1724.0
+#         },
+# 
+#         ...
+# 
+#         {
+#             "group": "logs-daily-iso",
+#             "logfile": "INFO: install_log = /var/tmp/kstest-driverdisk-disk-kargs.2022_03_04-02_42_13._25nhiki/virt-install.log",
+#             "start_time": 1646379735.0,
+#             "errors": [
+#                 "ERROR: Install failed: Test timed out",
+#                 "ERROR: ERROR: Image creation failed: Test timed out"
+#             ],
+#             "result": "INFO: RESULT:driverdisk-disk-kargs:ea89963f4984:FAILED:Test timed out",
+#             "success": false,
+#             "name": "driverdisk-disk-kargs",
+#             "end_time": 1646383341.0,
+#             "elapsed_time": 3606.0
+#         }
+# ]
+
+import argparse
+import json
+import os
+import re
+from time import strptime, mktime
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Convert kstest logs to JSON")
+    parser.add_argument("--group", required=True, help="Name of test group")
+    parser.add_argument("--output", help="Output filename (default is stdout)")
+    parser.add_argument("LOGFILE", type=os.path.abspath, help="File to convert")
+
+    return parser.parse_args()
+
+# Skip lines that are not from tests, or that should be ignored
+def skip(line):
+    if "apply-ksappend" in line:
+        return True
+
+    # Ignore the virt-install cmdline
+    if "virt-install [" in line:
+        return True
+
+    return False
+
+def main(args):
+    # Read the file, a line at a time
+    # Start by ignoring lines
+    capture_summary = False
+    tests = []
+    entry = {}
+    stats = {"total": 0, "pass": 0, "fail": 0}
```

Comment:
What is the purpose of this variable? I don't see it used anywhere.

Reply:
Ah, right, I dropped that from the json since it's all a flat list and easy enough to add up with when parsing the file.